### PR TITLE
Turn repository into a cookiecutter

### PR DESCRIPTION
**Pull request by jpmartinspt**
## Done
Turned this repo into a cookiecutter to make it really easy to setup a new flask site.

## QA
- `pip3 install -U cookiecutter`
- `cookiecutter https://github.com/jpmartinspt/website-boilerplate.git --checkout cookiecutter`

### CODE REVIEW

File: cookiecutter.json [hunk 1]
```diff
@@ -0,0 +1,8 @@
+{
+  "project_name": "Website",
+  "project_slug": "{{ cookiecutter.project_name.lower().replace(' ', '_').replace('-', '_') }}",
```
Comment (jkfran) [inline cookiecutter.json]: @jpmartinspt Why "-" to "_"? we have a lot of repos with "-"

File: {{cookiecutter.project_slug}}/requirements.txt [hunk 2]
```diff
@@ -1,2 +1 @@
 canonicalwebteam.flask-base==0.3.3
-gunicorn[gevent]
```
Comment (jkfran) [inline {{cookiecutter.project_slug}}/requirements.txt:2]: @jpmartinspt Why we don't need this library?

**Comment by jkfran** [comment]
Hey @jpmartinspt there is something wrong doing the QA:
```
cookiecutter https://github.com/jpmartinspt/website-boilerplate.git --checkout cookiecutter
project_name [Website]: fran-test
project_slug [fran_test]:          
project_port [8000]: 
github_username [canonical-web-and-design]: 
github_repository [https://github.com/canonical-web-and-design/fran_test]: 
version [0.1.0]: 
Traceback (most recent call last):
  File "/usr/bin/cookiecutter", line 11, in <module>
    load_entry_point('cookiecutter==1.6.0', 'console_scripts', 'cookiecutter')()
  File "/usr/lib/python3/dist-packages/click/core.py", line 764, in __call__
    return self.main(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/click/core.py", line 717, in main
    rv = self.invoke(ctx)
  File "/usr/lib/python3/dist-packages/click/core.py", line 956, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/usr/lib/python3/dist-packages/click/core.py", line 555, in invoke
    return callback(*args, **kwargs)
  File "/usr/lib/python3/dist-packages/cookiecutter/cli.py", line 120, in main
    password=os.environ.get('COOKIECUTTER_REPO_PASSWORD')
  File "/usr/lib/python3/dist-packages/cookiecutter/main.py", line 94, in cookiecutter
    output_dir=output_dir
  File "/usr/lib/python3/dist-packages/cookiecutter/generate.py", line 364, in generate_files
    generate_file(project_dir, infile, context, env)
  File "/usr/lib/python3/dist-packages/cookiecutter/generate.py", line 166, in generate_file
    tmpl = env.get_template(infile_fwd_slashes)
  File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 830, in get_template
    return self._load_template(name, self.make_globals(globals))
  File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 804, in _load_template
    template = self.loader.load(self, name, globals)
  File "/usr/lib/python3/dist-packages/jinja2/loaders.py", line 125, in load
    code = environment.compile(source, name, filename)
  File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 591, in compile
    self.handle_exception(exc_info, source_hint=source_hint)
  File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 780, in handle_exception
    reraise(exc_type, exc_value, tb)
  File "/usr/lib/python3/dist-packages/jinja2/_compat.py", line 37, in reraise
    raise value.with_traceback(tb)
  File "./run", line 177, in template
  File "/usr/lib/python3/dist-packages/jinja2/environment.py", line 497, in _parse
    return Parser(self, source, name, encode_filename(filename)).parse()
  File "/usr/lib/python3/dist-packages/jinja2/parser.py", line 901, in parse
    result = nodes.Template(self.subparse(), lineno=1)
  File "/usr/lib/python3/dist-packages/jinja2/parser.py", line 875, in subparse
    add_data(self.parse_tuple(with_condexpr=True))
  File "/usr/lib/python3/dist-packages/jinja2/parser.py", line 620, in parse_tuple
    args.append(parse())
  File "/usr/lib/python3/dist-packages/jinja2/parser.py", line 432, in parse_expression
    return self.parse_condexpr()
  File "/usr/lib/python3/dist-packages/jinja2/parser.py", line 437, in parse_condexpr
    expr1 = self.parse_or()
  File "/usr/lib/python3/dist-packages/jinja2/parser.py", line 450, in parse_or
    left = self.parse_and()
  File "/usr/lib/python3/dist-packages/jinja2/parser.py", line 459, in parse_and
    left = self.parse_not()
  File "/usr/lib/python3/dist-packages/jinja2/parser.py", line 470, in parse_not
    return self.parse_compare()
  File "/usr/lib/python3/dist-packages/jinja2/parser.py", line 474, in parse_compare
    expr = self.parse_math1()
  File "/usr/lib/python3/dist-packages/jinja2/parser.py", line 496, in parse_math1
    left = self.parse_concat()
  File "/usr/lib/python3/dist-packages/jinja2/parser.py", line 507, in parse_concat
    args = [self.parse_math2()]
  File "/usr/lib/python3/dist-packages/jinja2/parser.py", line 517, in parse_math2
    left = self.parse_pow()
  File "/usr/lib/python3/dist-packages/jinja2/parser.py", line 528, in parse_pow
    left = self.parse_unary()
  File "/usr/lib/python3/dist-packages/jinja2/parser.py", line 546, in parse_unary
    node = self.parse_primary()
  File "/usr/lib/python3/dist-packages/jinja2/parser.py", line 583, in parse_primary
    self.fail("unexpected '%s'" % describe_token(token), token.lineno)
  File "/usr/lib/python3/dist-packages/jinja2/parser.py", line 59, in fail
    raise exc(msg, lineno, self.name, self.filename)
jinja2.exceptions.TemplateSyntaxError: unexpected '.'
  File "./run", line 177
    if ! docker inspect -f {{.State.Running}} ${db_container} &>/dev/null; then
```

File: {{cookiecutter.project_slug}}/requirements.txt (hunk 2 above)
Comment (jpmartinspt) [reply {{cookiecutter.project_slug}}/requirements.txt:2]: flask-base install talisker and talisker brings gunicorn.

**Comment by jpmartinspt** [comment]
@jkfran should work now

File: cookiecutter.json (hunk 1 above)
Comment (jpmartinspt) [reply cookiecutter.json]: mistake! updated now..

**Review by jkfran — APPROVED** [review]

**Review by tbille — APPROVED** [review]
Seems fine too me.
Just a question: is there a reason we don't want the `yarn.lock` file?

**Comment by jpmartinspt** [comment]
@tbille I think it should be generated the first time we run the project so we get the highest possible package versions at the time we create the project, based on the constraints defined in package.json.

**Pull request merged**